esmvaltool/preprocessor/_derive/_shared.py

```diff
@@ -1,6 +1,64 @@
 """Auxiliary derivation functions used for multiple variables."""
 
+import logging
+
+import dask.array as da
 import iris
+from iris import Constraint
+
+logger = logging.getLogger(__name__)
+
+
+def _get_fx_cube(cubes, standard_name, fx_land_name, fx_sea_name=None):
```

Comment:
This function contains almost the same code twice, requires unnecessary arguments and it returns something complicated. Maybe you could replace it with a more simple function called something like `_get_land_fraction(cubes, standard_name)` that returns a dask array with the land fraction or None if it cannot be determined from either the sftlf or  sftof? The other functions using it can either use the resulting array directly or invert it if they need the sea fraction instead.